advisories/a99049.md

```diff
@@ -0,0 +1,37 @@
+---
+title: Technical Advisory 99049
+advisory: A-99049
+summary: Non-admin SQL users with an authenticated HTTP session could download statement diagnostic bundles given a bundle URL from the DB Console or SQL shell with a valid HTTP session cookie.
+toc: true
+affected_versions: v21.2.x, v22.1.0 to v22.1.16, v22.2.0 to v22.2.6
+advisory_date: 2023-03-27
+docs_area: releases
+---
+
+Publication date: {{ page.advisory_date | date: "%B %e, %Y" }}
+
+## Description
+
+This is a data leak/privilege escalation of statement diagnostic bundles by non-admin users. We expose the ability to download a statement diagnostic bundle over HTTP via a URL provided either by the [DB Console](../{{site.versions["stable"]}}/ui-statements-page.html#activate-diagnostics-collection-and-download-bundles) or the [SQL shell](../{{site.versions["stable"]}}/explain-analyze.html#debug-option). This URL is accessible to any logged-in user, but should be accessible only to [admin users](../{{site.versions["stable"]}}/security-reference/authorization.html#admin-role). This bundle contains unredacted query execution data including query text and the values of variables used within.
+
+A user would be vulnerable to this escalation if they have generated a statement diagnostic bundle via the DB Console or the SQL shell and shared the URL with someone without the privileges to view. Or, an attacker guessed the statement diagnostic bundle ID that is at the end of a URL, which is possible to guess or brute force because it is a combination of timestamp and node ID. The zip file that the attacker downloads would contain the unredacted query execution details.
+
+This vulnerability affects all of the following CockroachDB versions with this feature: v21.2.x, v22.1.0 to v22.1.16, v22.2.0 to v22.2.6.
+
+## Statement
+
+This is resolved in CockroachDB by PR [#99051](https://github.com/cockroachdb/cockroach/pull/99051), which fixes privilege escalation by correctly checking the SQL user’s privileges before granting access to the statement diagnostic bundle.
+
+The fix has been applied to maintenance releases of CockroachDB: [v22.1.17](../releases/v22.1.html) and [v22.2.7](../releases/v22.2.html).
+
+This public issue is tracked by [#99049](https://github.com/cockroachdb/cockroach/issues/99049).
+
+## Mitigation
+
+Users of CockroachDB v21.2, v22.1, and v22.2 are encouraged to upgrade to v22.1.17, v22.2.7, or a later version. They can alternatively block HTTP access to the statement diagnostics download endpoint on their cluster via an HTTP proxy.
```

Comment:
Wonder if it's worth mentioning that if you never generate a bundle, or delete the bundles by dropping rows from `system.statement_diagnostics` to remove existing generated bundles you can also protect against any access.

Reply:
Something like this @dhartunian? I can add to the end of the Mitigation section?

> Users can also protect against access if they do not generate statement diagnostics bundles, or delete the bundles by dropping rows from `system.statement_diagnostics` to remove existing generated bundles.